vdsm_hooks/log_firmware/before_vm_start.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+
+import os
+import re
+import traceback
+
+from vdsm.hook import hooking
+
+
+def main():
+    log_firmware = os.environ.get('log_firmware')
+    hooking.log(f'log_firmware: "{log_firmware}"')
+    if log_firmware == 'off':
+        return
+
+    log_firmware_vm_regexp = os.environ.get('log_firmware_vm_regexp')
+    hooking.log(f'log_firmware_vm_regexp: "{log_firmware_vm_regexp}"')
+
+    domxml = hooking.read_domxml()
+    vm_name = domxml.getElementsByTagName('name')[0].firstChild.data
+    hooking.log(f'vm_name: {vm_name}')
+
+    if (
+        log_firmware != 'on'
+        and (
+            not log_firmware_vm_regexp
+            or not re.match(log_firmware_vm_regexp, vm_name)
```

Comment:
is / is not None would be idiomatic but no big deal.

Reply:
I admit I never heard that this is considered idiomatic.

See e.g. the start of [Match Objects](https://docs.python.org/3/library/re.html#match-objects):
```
Match objects always have a boolean value of True. Since [match()](https://docs.python.org/3/library/re.html#re.Pattern.match) and [search()](https://docs.python.org/3/library/re.html#re.Pattern.search) return None when there is no match, you can test whether there was a match with a simple if statement:

match = re.search(pattern, string)
if match:
    process(match)
```

Searching, rather naively, the current master branch of vdsm, I see a few such cases, but not many:
```
[ybardavi@ybardavi vdsm]$ grep -rw 'if .*\.match\>' * | wc -l
16
[ybardavi@ybardavi vdsm]$ grep -rw 'if .*\.match\>.*None' * | wc -l
5
```

Reply:
IIUC you refer to the case where a user is setting log_firmware_vm_regexp to an empty string.

It's indeed not documented in the README, but I actually think the code does the sane thing and ignores it, instead of matching all possible VM names. I did consider this case when writing and deciding on `not log_firmware_vm_regexp` instead of `log_firmware_vm_regexp is None`.

I obviously agree with you that as a programmer, I should always be aware of the difference and consciously pick one or the other. Not always this is clear from the context/comments, sorry...

Reply:
> IIUC you refer to the case where a user is setting log_firmware_vm_regexp to an empty string.

Yes.

> It's indeed not documented in the README, but I actually think the code does the sane thing and ignores it, instead of matching all possible VM names. I did consider this case when writing and deciding on `not log_firmware_vm_regexp` instead of `log_firmware_vm_regexp is None`.

OK.

I already approved this PR, waiting whether @tinez would like to add anything.